File: csharp/BookStore/src/BookStore/GiftWrapping.cs
namespace BookStore
{
    public class GiftWrapping : AbstractItem
    {
        public GiftWrapping(AbstractItem item) : base($"Gift wrapping for {item.GetName()}")
        {
        }

        public override void PutIntoMyCart(Cart cart)
        {
            cart.Add(GetName());
        }

        protected override void CalculateDeliveryCost(Cart cart, int gramms)
        {
            // no delivery cost
            MarkReadyToDeliver();
        }

        protected override int ItemWeight()
        {
            return 5;
        }
    }
}

File: cpp/book_store-cmake_catch2/src/GiftWrapping.cpp
//
// Created by mille on 28-Sep-22.
//

#include "GiftWrapping.h"

namespace bookstore {
  GiftWrapping::GiftWrapping(bookstore::AbstractItem& item) : AbstractItem("Gift wrapping for " + item.getName()){

  }

  void GiftWrapping::putIntoMyCart(bookstore::Cart& cart) {
    cart.add(getName());
  }

  void GiftWrapping::calculateDeliveryCost(bookstore::Cart& cart, int gramms) {
    markReadyToDeliver();
  }

  int GiftWrapping::itemWeight() {
    return 5;
  }
}